wsapi/server.go

```diff
@@ -108,10 +109,26 @@ func IDInjector(server *Server) Middleware {
 	}
 }
 
+func PanicRecovery() Middleware {
+	return func(f http.HandlerFunc) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			defer func() {
+				if r := recover(); r != nil {
```

Comment:
nit but you have 2 variables with the same name that mask one another: `r *http.Request` and `r := recover()`, can you make explicit different variable names?